exercises/exercise-05/index.ts

```diff
@@ -64,11 +64,18 @@ function logPerson(person: Person) {
     );
 }
 
-function filterPersons(persons: Person[], personType: string, criteria: unknown): unknown[] {
+function getObjectKeys<T extends object>(o: T): (keyof T)[] {
```

Comment:
The instructions for this are:

> Implement a function `getObjectKeys()` which returns proper type
> for any argument given, so that you don't need to cast it.

"More convenient", sure, but "proper" is too strong here, `string[]` really is the correct type for the reasons outlined in [this StackOverflow answer](https://stackoverflow.com/questions/55012174/why-doesnt-object-keys-return-a-keyof-type-in-typescript) and [this TS issue](https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208).

There's also a discussion of this in _Effective TypeScript_ (Item 54). I'm happy to get you a free copy if you'd like.